cmd/phlare/main.go

```diff
@@ -16,6 +16,10 @@ import (
 	_ "github.com/grafana/phlare/pkg/util/build"
 )
 
+// testMode is used to test flag parsing.
+// We don't want to exit the process when running tests.
+var testMode = false
```

Comment:
I would move this as a private global to `pkg/cfg/dynamic.go`.

Then I would have a public method `SetTestMode(bool)` in `pkg/cfg/dynamic_test.go`. This doesn't pollute the arguments. How do you feel about that?

Reply:
Yeah. It sounds good! Thanks for your suggestion!